aip/client-libraries/4222.md

```diff
@@ -154,6 +154,31 @@ If none of the routing parameters matched their respective fields, the routing h
 Much like URL parameters, if there is more than one key-value pair to be sent, the `&`
 character is used as the separator.
 
+### `path_template` syntax
+
+As seen in the above examples, the `path_template` can use a variety of symbols
+that are interpreted by code generators during conversion to regular expression.
+The syntax for these symbols is as follows:
+
+- Wildcards:
+  - `*`: Corresponds to 1 or more non-`/` symbols. The regex describing it is
```

Comment:
> Does it mean we want to support * or ** anywhere in the path template? e.g. Something like {bucket=pro*jects/**buckets/*}?

This is a great question, one I don't think I have an answer to. I agree that perhaps a more restrictive approach is better to start. That said, I'm not sure we _need_ to be so restrictive (with all the regulation that comes with that) that the wildcards can only be used in X Y and Z scenarios. I'd like to hear what others have to say, but I will start thinking of some verbiage for how we might scope its use.

Reply:
I agree; I think we need to be explicitly restrictive, especially since it better expresses the current intent.

Reply:
You're right. I've added a Delimiters section that names `/` and `:` as acceptable segment delimiters (since they technically are in `google.api.http` paths, with `:` used to separate custom method names) and a **must** requirement to the wildcards that a delimiter precede the wildcard.

I'll resolve this and y'all can open comments on those new/added sections.